assets/src/components/detourListPage.tsx

```diff
@@ -1,21 +1,52 @@
-import React, { useCallback, useState } from "react"
+import React, { useCallback, useEffect, useState } from "react"
 import { DetoursTable } from "./detoursTable"
 import userInTestGroup, { TestGroups } from "../userInTestGroup"
 import { Button } from "react-bootstrap"
 import { PlusSquare } from "../helpers/bsIcons"
 import { DetourModal } from "./detours/detourModal"
-import { fetchDetours } from "../api"
+import { fetchDetour, fetchDetours } from "../api"
 import { useApiCall } from "../hooks/useApiCall"
 import { isOk } from "../util/result"
+import { isValidSnapshot } from "../util/isValidSnapshot"
+import { createDetourMachine } from "../models/createDetourMachine"
+import { Snapshot } from "xstate"
 
 export const DetourListPage = () => {
   const [showDetourModal, setShowDetourModal] = useState(false)
+  const [detourId, setDetourId] = useState<number | undefined>()
+  const [stateOfDetourModal, setStateOfDetourModal] =
+    useState<Snapshot<unknown> | null>()
 
   const { result } = useApiCall({
     apiCall: useCallback(async () => fetchDetours(), []),
   })
   const detours = result && isOk(result) && result.ok
 
+  const { result: detourResult } = useApiCall({
+    apiCall: useCallback(
+      async () => (detourId ? fetchDetour(detourId) : undefined),
+      [detourId]
+    ),
+  })
+  const detour =
+    detourResult && isOk(detourResult) ? detourResult.ok : undefined
+
+  useEffect(() => {
+    const state = isValidSnapshot(createDetourMachine, detour?.state)
+    if (isOk(state)) setStateOfDetourModal(state.ok)
+  }, [detour])
```

Comment:
suggestion(non-blocking): you don't have to do this, but since our only usage of the detour result is the ok state, you could condense this into a single `useEffect` via the `useApiCall` hook

```suggestion
  const { result: stateOfDetourModal } = useApiCall({
    apiCall: useCallback(
      async () => {
        if (detourId === undefined) { return undefined }

        const detourResponse = await fetchDetour(detourId)

        if (isErr(detourResponse)) { return undefined }

        const snapshot = isValidSnapshot(createDetourMachine, detourResponse.ok.state)
        if (isErr(snapshot)) { return undefined }

        return snapshot.ok
      },
      [detourId]
    ),
  })
```